### PR TITLE
xds: Update CEL matcher proto status

### PR DESCRIPTION
**Pull request by tyxia**
- CEL proto and matcher have been used in production over 1 year, remove work-in-progress

- Also add the extension category.

### CODE REVIEW

File: xds/type/matcher/v3/cel.proto [hunk 1]
```diff
@@ -30,8 +27,7 @@ option (xds.annotations.v3.file_status).work_in_progress = true;
 // Refer to :ref:`Unified Matcher API <envoy_v3_api_msg_.xds.type.matcher.v3.Matcher>` documentation
 // for usage details.
 //
-// [#comment:TODO(sergiitk): Link HttpAttributesMatchInput + usage example.]
```
Comment (adisuissa) [inline xds/type/matcher/v3/cel.proto:33]: @sergiitk is removing the TODOs make sense to you, or should more examples be added?

**Review by adisuissa — COMMENTED** [review]
Overall LGTM, waiting on question.

File: xds/type/matcher/v3/cel.proto (hunk 1 above)
Comment (tyxia) [reply xds/type/matcher/v3/cel.proto:33]: Yeah, i have thought about this.  I feel examples is probably overkill here. Looking through files in xds/type/matcher/v3, I only see string.proto has simple examples.

But I can keep this TODO if @sergiitk want.

Comment (sergiitk) [reply xds/type/matcher/v3/cel.proto:33]: In this todo, I've probably meant `HttpAttributesCelMatchInput`, not `HttpAttributesMatchInput`.  Regarding what to link, I found this: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/matching/matching_api.html#custom-matching-algorithms. I think it'll fit perfectly here.

File: xds/type/matcher/v3/cel.proto
Comment (sergiitk) [inline xds/type/matcher/v3/cel.proto:1]: @tyxia - would it make sense to remove work in progress from `xds.type.matcher.v3.HttpAttributesCelMatchInput`  as well, in the same PR? They're pretty much always used together. https://github.com/cncf/xds/blob/024c85f92f20cab567a83acc50934c7f9711d124/xds/type/matcher/v3/http_inputs.proto#L16-L27

**Review by sergiitk — COMMENTED** [review]
Could you also please fix the title to say something like "stabilize CelMatcher`. There a few `cel.proto` files in this repo, we want to be more clear which one we update.

File: xds/type/matcher/v3/cel.proto (hunk 1 above)
Comment (tyxia) [reply xds/type/matcher/v3/cel.proto:33]: Unified Matcher API is already linked  in the comment above. Also, `envoy.matching.matchers.cel_matcher` should provides enough information for customer to navigate.

File: xds/type/matcher/v3/cel.proto
Comment (tyxia) [reply xds/type/matcher/v3/cel.proto:1]: Sure Done. PTAL.

File: xds/type/matcher/v3/cel.proto (hunk 1 above)
Comment (sergiitk) [reply xds/type/matcher/v3/cel.proto:33]: Up to you - but the Unified Matcher API link above points to the generated proto page (https://www.envoyproxy.io/docs/envoy/latest/xds/type/matcher/v3/matcher.proto#envoy-v3-api-msg-xds-type-matcher-v3-matcher), while https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/matching/matching_api.html#custom-matching-algorithms is more of a user-friendly page that explains how to use the Matcher API. 

I'm fine with the doc as it is though.

**Review by sergiitk — APPROVED** [review]
Looks great, thank you!

**Comment by sergiitk** [comment]
@tyxia - the CI is failing because go protos needs to be regenerated and committed, see  https://github.com/cncf/xds/blob/main/DEVELOPER.md

```sh
bazel build //...
tools/generate_go_protobuf.py
```

**Comment by tyxia** [comment]
> @tyxia - the CI is failing because go protos needs to be regenerated and committed, see https://github.com/cncf/xds/blob/main/DEVELOPER.md
> 
> ```shell
> bazel build //...
> tools/generate_go_protobuf.py
> ```

Good to know. Thank you!

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Pull request merged**